packages/utils/src/validators.ts

```diff
@@ -119,74 +125,220 @@ export function isSessionStruct(input: any): input is SessionTypes.Struct {
   return input?.topic;
 }
 
-export function isValidRequiredNamespaceBody(
-  input: any,
-): input is ProposalTypes.BaseRequiredNamespace {
-  const { methods, events, chains } = input;
-  let validChains = true;
-  const validRequiredNamespace =
-    isValidArray(methods) && isValidArray(events) && isValidArray(chains);
-  if (validRequiredNamespace) {
-    chains.forEach((chain: string) => {
-      if (!isValidChainId(chain, false)) validChains = false;
-    });
+export function isValidController(input: any, method: string) {
+  let error: ErrorObject = null;
+  if (!isValidString(input?.publicKey, false)) {
+    error = getInternalError(
+      "MISSING_OR_INVALID",
+      `${method} controller public key should be a string`,
+    );
   }
 
-  return validRequiredNamespace && validChains;
+  return error;
 }
 
-export function isValidRequiredNamespaces(
-  input: any,
-  optional: boolean,
-): input is ProposalTypes.RequiredNamespaces {
-  let valid = false;
-  if (optional && !input) return true;
-  else if (input && isValidObject(input)) {
-    valid = true;
-    Object.values(input).forEach((namespace: any) => {
-      if (!isValidRequiredNamespaceBody(namespace)) valid = false;
-      if (valid && namespace?.extension) {
-        if (!isValidArray(namespace.extension)) valid = false;
-        namespace.extension.forEach((extension: any) => {
-          if (!isValidRequiredNamespaceBody(extension)) valid = false;
-        });
-      }
-    });
+export function isValidExtension(namespace: any, method: string) {
+  let error: ErrorObject = null;
+  if (!isUndefined(namespace?.extension)) {
+    if (!isValidArray(namespace.extension) || !namespace.extension.length) {
+      error = getInternalError(
+        "MISSING_OR_INVALID",
+        `${method} extension should be an array of namespaces, or omitted`,
+      );
+    }
+  }
+
+  return error;
+}
+
+export function isValidNamespaceMethodsOrEvents(input: any): input is string {
+  let valid = true;
+  if (isValidArray(input)) {
+    if (input.length) {
+      input.forEach((item: any) => {
```

Comment:
Think this can also be simplified with `every`, which is doing basically the same thing under the hood:

```suggestion
      valid = input.every((item: any) => isValidString(item, false))
```